core/src/main/scala/fs2/io/ssh/Client.scala

```diff
@@ -88,7 +93,20 @@ final class Client[F[_]: Concurrent: ContextShift] private (client: SshClient) {
           }
       }
 
-      success <- Resource.liftF(fromFuture(F.delay(session.auth())))
+      success <- Resource.liftF(fromFuture(F.delay(session.auth()))).attempt
+
+      _ <- Resource liftF {
+        success match {
+          case Left(_: SshException) =>
+            FR.raise(Error.Authentication)
```

Comment:
Maybe include the message from the `SshException`?

Reply:
Here's the exception:

```
[error]    org.apache.sshd.common.SshException: No more authentication methods available (ClientUserAuthService.java:331)
[error] org.apache.sshd.client.session.ClientUserAuthService.tryNext(ClientUserAuthService.java:331)
[error] org.apache.sshd.client.session.ClientUserAuthService.processUserAuth(ClientUserAuthService.java:267)
[error] org.apache.sshd.client.session.ClientUserAuthService.process(ClientUserAuthService.java:214)
[error] org.apache.sshd.common.session.helpers.AbstractSession.doHandleMessage(AbstractSession.java:435)
[error] org.apache.sshd.common.session.helpers.AbstractSession.handleMessage(AbstractSession.java:362)
[error] org.apache.sshd.common.session.helpers.AbstractSession.decode(AbstractSession.java:1207)
[error] org.apache.sshd.common.session.helpers.AbstractSession.messageReceived(AbstractSession.java:323)
[error] org.apache.sshd.common.session.helpers.AbstractSessionIoHandler.messageReceived(AbstractSessionIoHandler.java:63)
[error] org.apache.sshd.netty.NettyIoSession.channelRead(NettyIoSession.java:254)
[error] org.apache.sshd.netty.NettyIoSession$Adapter.channelRead(NettyIoSession.java:298)
[error] io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:374)
[error] io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:360)
[error] io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:352)
[error] io.netty.handler.logging.LoggingHandler.channelRead(LoggingHandler.java:241)
[error] io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:374)
[error] io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:360)
[error] io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:352)
```

It's kind of not useful in this context. :-(